server/auth.go

```diff
@@ -338,6 +338,9 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 		ao   bool // auth override
 	)
 	s.mu.Lock()
+	if len(s.trustedKeys) == 0 {
```

Comment:
I wonder if we should either clear it directly when parsing the connect, say before this line: https://github.com/nats-io/nats-server/blob/6d06e393eeb9bce911899f18ae887c460fa48e8d/server/client.go#L1544 (but then would need to have server lock somewhere to check for trusted keys, unless it is immutable), or then we should do that under c's lock.

Reply:
Moved the code. s.trustedKeys is set right after newserver. 
After that the pointer is not touched. (this is why i changed the check to == nil)
So I assume the server lock is not needed.